edward/models/distributions.py

```diff
@@ -70,45 +73,80 @@ def sample(self, size=1):
 
         Returns
         -------
-        tf.Tensor, list
-            A tensor concatenating sample outputs of tensors and
-            placeholders. The list used to form the tensor is also
-            returned so that other procedures can feed values into the
-            placeholders.
+        list or tf.Tensor
+            If more than one layer, a list of tensors, one for each
+            layer. If one layer, a single tensor. If a layer requires
+            SciPy to sample, its corresponding tensor is a
+            tf.placeholder.
         """
         samples = []
         for layer in self.layers:
             if layer.sample_tensor:
                 samples += [layer.sample(size)]
             else:
-                samples += [tf.placeholder(tf.float32, (size, layer.num_vars))]
+                samples += [tf.placeholder(tf.float32, (size, ) + layer.shape)]
 
-        return tf.concat(1, samples), samples
+        if len(samples) == 1:
+            samples = samples[0]
+
```

Comment:
It's useful to distinguish between the container of distribution objects and the shape of their samples. The above is unrelated to the number of drawn samples. It's relates to whether the container only holds one layer or it holds multiple layers.